### PR TITLE
[GNA] Overload correction algorithm for Add (Renew PR11406)

### PR DESCRIPTION
**Pull request by nkogteva**
### Details:
 - *Added overload correction for Add operation inside GNA plugin*
 - *Added a test for overload correction case*

### Tickets:
 - *83490*

Original PR https://github.com/openvinotoolkit/openvino/pull/11406

### CODE REVIEW

File: src/plugins/intel_gna/frontend/scale_factor_calc.hpp [hunk 1]
```diff
@@ -220,71 +220,81 @@ static double calculateWeightsReducerFromDstStats(Quantization dst_quant) {
...
+        return true;
+    }
 
-        if (info.isFullyConnected() || info.isConvolution()) {
-            quantDataForInputLayer->_dst_quant.SetScale(newOutputScale);
-            quantDataForInputLayer->_weights_quant.SetScale(newOutputScale / quantDataForInputLayer->_src_quant.GetScale());
-            result = ScaleFactorUpdateResult(layer.get());
-            return true;
-        }
+    if (info.isFullyConnected() || info.isConvolution() || info.isPower()) {
```
Comment (sirzabek) [inline src/plugins/intel_gna/frontend/scale_factor_calc.hpp:262]: My general comment is that it's hard to follow changes in this PR without comments on what has changed and why (in code or in the ticket).
For example - how is this change connected with Add layer?

**Review by sirzabek — APPROVED** [review]

File: src/plugins/intel_gna/frontend/scale_factor_calc.hpp (hunk 1 above)
Comment (nkogteva) [reply src/plugins/intel_gna/frontend/scale_factor_calc.hpp:262]: > My general comment is that it's hard to follow changes in this PR without comments on what has changed and why (in code or in the ticket). For example - how is this change connected with Add layer?

I suppose changes are mostly connected to specific customer model, not just Add. For example, on your branch this change is present under commit "Accuracy fixes for PWL and Eltwise" https://github.com/sirzabek/openvino/commit/d6989b1a8bfac28a1aa1c47971f173fa1ea290c7
